### DISCUSSION TITLE
Ratio of normsys modifiers

### DISCUSSION THREAD

**Discussion by itsaklid** [scikit-hep/pyhf#2441] [Q&A]
Hello,

If I have a normsys modifier named `modifierA`, attached to a `Sample A`, is it possible to have the ratio $\frac{modifierB}{modifierA}$ attached to a `Sample B` ,  where `modifierB` is another normsys modifier with the current pyhf implementation ? 

Thanks a lot in advance!

**Comment by alexander-held** [comment]
For ratios there is a trick where you use the ratio itself as a modifier and then the denominator in addition as the second modifier. See https://github.com/scikit-hep/pyhf/discussions/2399#discussioncomment-7818657 for this approach when using `normfactor` modifiers.

I have not seen seen this used or thought about it for `normsys` modifiers, so not sure spontaneously if the exponential extrapolation (compared to the linear case for `normfactor`) makes this unsuitable. The modifiers still both multiply the effect of samples, which is what this approach mainly relies on.

**Reply by itsaklid** [reply]
Hi @alexander-held , thanks a lot for your answer!

Indeed I'm doing something along the lines of [#2399](https://github.com/scikit-hep/pyhf/discussions/2399#discussioncomment-7818657) for the `normfactor` modifiers which appear as ratios in my fit.

I tried to keep my question as generic as possible which hid some of the complexity of the situation I'm in. The subtle point is that the correlation of `modifierA` and `modifierB` with respective `histosys` modifiers might need to be taken into account. This is of course similar to what cabinetry implements with the `NormPlusShape` systematic, but due to case-specific reasons I'm disentangling the normalization from the shape effects at first, similar to what is envisioned in the [ #477](https://github.com/scikit-hep/cabinetry/issues/447) cabinetry issue and then manually correlating only certain shape to certain normalization effects (with some of the latter ones appearing in ratios).

If I define the ratio of the `normsys` modifiers beforehand, propagate their uncertainties correctly and introduce a new modifier for it, then it seems to me that the correlations with the respective `histosys` modifiers would not be taken into account properly. Would you agree with such a statement ?

Taking this a step further, the experimental implementation of custom modifiers in #1991 can handle only `normfactor` modifers as of now, is that right ? Or am I perhaps misinterpreting something ?

Thanks a lot for you time!

Cheers,
Ilias

**Reply by alexander-held** [reply]
This sounds like a very unusual setup to me and I think I would need to understand it in more detail to offer useful advice. At a high level I wonder if there might be a better way to parameterize the uncertainties for your model to simplify the setup. It is not obvious to me that `histosys` contributions would be problematic at the moment but that may just be my lack of understanding of what you are looking for.

Regarding #1991: I am not sure myself but I believe the scope was broader than just `normfactor` modifiers, @kratsg @lukasheinrich @matthewfeickert could you confirm? I have only tried it with `normfactor` myself thus far.
